plugins/restful/RestfulBase.php

```diff
@@ -815,7 +815,16 @@ public function process($path = '', array $request = array(), $method = \Restful
       $this->getRateLimitManager()->checkRateLimit($request);
     }
 
-    return $this->{$method_name}($path);
+    $return = $this->{$method_name}($path);
+
+    if (empty($request['__application']['rest_call'])) {
```

Comment:
> but we are only switching it back if the call is from cURL?

No, we only switch back inside an API call. We don't do it on cURL as:
1) The user might have logged in using the cookie login
2) There's no need since the session isn't saved, and the request will terminate "soon".